Docs: Add the `Optimizing build size` document (#16606)
ckeditor/ckeditor5: docs/getting-started/setup/optimizing-build-size.md

```js
// Import the core styles.
import '@ckeditor/ckeditor5-theme-lark/dist/index.css';
import '@ckeditor/ckeditor5-clipboard/dist/index.css';
import '@ckeditor/ckeditor5-core/dist/index.css';
import '@ckeditor/ckeditor5-engine/dist/index.css';
import '@ckeditor/ckeditor5-enter/dist/index.css';
import '@ckeditor/ckeditor5-paragraph/dist/index.css';
import '@ckeditor/ckeditor5-select-all/dist/index.css';
import '@ckeditor/ckeditor5-typing/dist/index.css';
import '@ckeditor/ckeditor5-ui/dist/index.css';
import '@ckeditor/ckeditor5-undo/dist/index.css';
import '@ckeditor/ckeditor5-upload/dist/index.css';
import '@ckeditor/ckeditor5-utils/dist/index.css';
import '@ckeditor/ckeditor5-watchdog/dist/index.css';
import '@ckeditor/ckeditor5-widget/dist/index.css';
```

Then, import the styles for the plugins that you use. For example, if you use the bold, italic and table features, you can import just the styles for those features:

```js
// Import the styles for the features that you use.
import '@ckeditor/ckeditor5-basic-styles/dist/index.css';
import '@ckeditor/ckeditor5-table/dist/index.css';
// ...
```

> When it comes to plugin styles, the rule of thumb is to import the styles from all the individual packages from which you import the editor features. For example, if you import the bold feature from `@ckeditor/ckeditor5-basic-styles/dist/index.js`, you should also import `@ckeditor/ckeditor5-basic-styles/dist/index.css`.

Reviewer: ```suggestion
Regarding plugin styles, the rule of thumb is to import the styles from all the individual packages from which you import the editor features. For example, if you import the bold feature from `@ckeditor/ckeditor5-basic-styles/dist/index.js`, you should also import `@ckeditor/ckeditor5-basic-styles/dist/index.css`.
```

You may notice that some plugin style sheets are empty. This is intentional, as some plugins do not have styles now, but may have them in the future. Adding the imports now will ensure that you do not accidentally miss some styles if that happens. Importing empty stylesheets does not increase the build size.

> 

Reviewer 2: Maybe would be good to also mention somewhere here about CSS that not only `index.css` can be imported but it's divided into `content` and `editor`? Sounds also like a potential optimization, WDYT?

Author: I was thinking the same thing, but it's already explained in a separate document and I don't want to duplicate it.

https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/getting-started/setup/css.html

Reviewer 2: Yes, thats true.

Maybe a link with short sentence to it could be a good idea? :thinking: 

Its always better to have it in one place (i mean all the necessary links that could help) - when someone will enter this article right from the search engine (like google) will reach all necessary information.

Reviewer 3: Maybe just link it, then?

Author: Done :heavy_check_mark:

import undoTranslations from '@ckeditor/ckeditor5-undo/dist/translations/<LANGUAGE>.js';
import uploadTranslations from '@ckeditor/ckeditor5-upload/dist/translations/<LANGUAGE>.js';
import widgetTranslations from '@ckeditor/ckeditor5-widget/dist/translations/<LANGUAGE>.js';
```

Then, import the translations for the plugins that you use. For example, if you use the `Bold`, `Italic` or `Table` features, you can only import the translations for those features:

```js
// Import the translations for the features that you use.
import basicStylesTranslations from '@ckeditor/ckeditor5-basic-styles/dist/translations/<LANGUAGE>.js';
import tableTranslations from '@ckeditor/ckeditor5-table/dist/translations/<LANGUAGE>.js';
```

As with styles, the rule of thumb is to import translations from all the individual packages from which you import the editor features. For example, if you import the `Bold` feature from the `@ckeditor/ckeditor5-basic-styles/dist/index.js` package, you should also import the translations from the `@ckeditor/ckeditor5-basic-styles/translations/<LANGUAGE>.js` file.

Some plugins may not have translations. In such cases, you do not need to import translations for them.

> ## The result

Reviewer 3: It is somewhat redundant, but not useless. I'd keep it.

> Let's see how the build size changes after applying the above optimizations to a sample project with classic editor and a variety of free and premium features and Polish translations.

Reviewer: ```suggestion
Let's see how the build size changes after applying the above optimizations to a sample project with the classic editor, free and premium features, plus Polish translations.
```

			'tableRow',
			'mergeTableCells'
		]
	},
	translations: [
		clipboardTranslations,
		coreTranslations,
		enterTranslations,
		selectAllTranslations,
		uiTranslations,
		undoTranslations,
		uploadTranslations,
		widgetTranslations,
		autoformatTranslations,
		basicStylesTranslations,
		blockQuoteTranslations,
		ckboxTranslations,
		ckfinderTranslations,
		headingTranslations,
		imageTranslations,
		indentTranslations,
		linkTranslations,
		listTranslations,
		mediaEmbedTranslations,
		tableTranslations,
		caseChangeTranslations,
		slashCommandTranslations,
	],
	language: 'pl'
} );
```
</details>

The build size of the project before and after the optimizations is as follows:

|                    	| Before optimization 	| After optimization 	| Improvement 	|
|--------------------	|---------------------	|--------------------	|-------------	|
| JavaScript         	| 1,248.02 kB         	| 1,029.01 kB        	| -17.55%     	|
| CSS                	| 340.45 kB           	| 172.84 kB          	| -49.23%     	|
| JavaScript gzipped 	| 343.56 kB           	| 288.84 kB          	| -15.93%     	|
| CSS gzipped        	| 46.16 kB            	| 28.71 kB           	| -37.80%     	|

> Thanks to the above optimizations, we were able to reduce the total build size by `386.62 kB` (`72.17 kB` gzipped), which is ~80% of the original size. These results will vary depending on the features you use.

Reviewer 3: I can't really admit I get the "386kB out of 1.2MB makes 80%" math.

Author: The combined size of JavaScript and CSS has been reduced by 20 to 25% (from `1,248.02 + 340.45` to `1,029.01 + 172.84`).

I think we need to rephrase this sentence to make it clear.

Author: ```suggestion
Thanks to the above optimizations, we were able to reduce the total build size (JavaScript + CSS) by `386.62 kB` (`72.17 kB` gzipped), which is ~75-80% of the original size. These results will vary depending on the features you use.
```
